nixos/modules/services/networking/adguardhome.nix

```diff
@@ -150,6 +150,11 @@ in
       preStart = optionalString (cfg.settings != null) ''
         if    [ -e "$STATE_DIRECTORY/AdGuardHome.yaml" ] \
            && [ "${toString cfg.mutableSettings}" = "1" ]; then
+          # First run a schema_version update on the existing configuration
+          # This ensures that both the new config and the existing one have the same schema_version
+          # Note: --check-config has the side effect of modifying the file at rest!
+          ${pkgs.adguardhome}/bin/adguardhome -c "$STATE_DIRECTORY/AdGuardHome.yaml" --check-config
```

Comment:
`${pkgs.adguardhome}/bin/adguardhome` -> `${lib.getExe adguardhome}`?

Reply:
Also fixed the other occurrences. Tests pass for me on `x86_64-linux`.